nengo_loihi/hardware/interface.py

```diff
@@ -215,13 +215,26 @@ def _chip2host_monitor(self, probes_receivers):
     def _chip2host_snips(self, probes_receivers):
         assert self.host_socket_connected
 
+        expected_bytes = 4 * self.nengo_io_c2h_count
+        n_waits = 0  # number of times we've had to wait for more data
+
         recv_size = 4096  # python docs recommend small power of 2, e.g. 4096
         received = self.host_socket.recv(recv_size)  # blocking recv call
         data = received
-        while len(received) == recv_size:
-            received = self.host_socket.recv(recv_size, flags=socket.MSG_DONTWAIT)
-            data += received
-        assert len(data) == 4 * self.nengo_io_c2h_count
+        while len(data) < expected_bytes and n_waits < 10:
```

Comment:
It may be helpful in the future to add a comment on why `n_waits < 10`, even if it is chosen at random

Reply:
It might be good to have a class-level variable for this. Each wait is 1 ms, so we could have `snip_max_interpacket_wait_ms` or something.